mpi-proxy-split/mpi_plugin.cpp

```diff
@@ -1122,6 +1165,36 @@ mpi_plugin_event_hook(DmtcpEvent_t event, DmtcpEventData_t *data)
   }
 }
 
+void
+global_p2p_communication_barrier()
+{
+  if (global_p2p_communication == 1 ||
```

Comment:
Does it have race condition between checkpoint thread and user application threads? Should we use lock to protect global_p2p_communication access?